### PR TITLE
[teleport-update] Show warning instead of return error for link/unlink

### PR DESCRIPTION
**Pull request by vapopov**
In this PR added fix to make `link-package` and `unlink-package` less strict to have `systemd` installed, related to deb, rpm installation tests

https://github.com/gravitational/teleport.e/actions/runs/11935999685/job/33269358902

### CODE REVIEW

File: lib/autoupdate/agent/updater.go [hunk 1]
```diff
@@ -749,7 +749,9 @@ func (u *Updater) LinkPackage(ctx context.Context) error {
 	} else if err != nil {
 		return trace.Errorf("failed to link system package installation: %w", err)
 	}
-	if err := u.Process.Sync(ctx); err != nil {
+	if err := u.Process.Sync(ctx); errors.Is(err, ErrNotSupported) {
+		u.Log.WarnContext(ctx, "SystemD is not installed, skip sync")
```
Comment (sclevine) [inline lib/autoupdate/agent/updater.go:753]:
```suggestion
		u.Log.WarnContext(ctx, "Systemd is not installed. Skipping sync.")
```
Just for consistency with the other logs. (SystemD is probably more correct.)

File: lib/autoupdate/agent/updater.go [hunk 2]
```diff
@@ -762,7 +764,9 @@ func (u *Updater) UnlinkPackage(ctx context.Context) error {
 	if err := u.Installer.UnlinkSystem(ctx); err != nil {
 		return trace.Errorf("failed to unlink system package installation: %w", err)
 	}
-	if err := u.Process.Sync(ctx); err != nil {
+	if err := u.Process.Sync(ctx); errors.Is(err, ErrNotSupported) {
+		u.Log.WarnContext(ctx, "SystemD is not installed, skip sync")
```
Comment (sclevine) [inline lib/autoupdate/agent/updater.go:768]:
```suggestion
		u.Log.WarnContext(ctx, "Systemd is not installed. Skipping sync.")
```
Just for consistency with the other logs. (SystemD is probably more correct.)

**Review by sclevine — APPROVED** [review]
(Reminder to add test)

**Review by sclevine — APPROVED** [review]

**Review by hugoShaka — APPROVED** [review]

**Comment by vapopov** [comment]
@bl-nero @ryanclark I would appreciate your review of this small PR

**Review by bl-nero — APPROVED** [review]

**Pull request merged**